src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -56,64 +47,95 @@ val jsonValueColumnsSql = PostgresDataTables.dataColumns.entries
  * 3. partition(s) (array)
  *
  */
-fun buildPreparableFiltersSqlForLinkedEntities(
+fun buildPreparableFiltersSql(
         startIndex: Int,
         propertyTypes: Map<UUID, PropertyType>,
         propertyTypeFilters: Map<UUID, Set<Filter>>,
+        metadataOptions: Set<MetadataOption>,
+        linking: Boolean,
         idsPresent: Boolean,
-        partitionsPresent: Boolean,
-        selectOriginIds: Boolean = false
+        partitionsPresent: Boolean
 ): Pair<String, Set<SqlBinder>> {
     val filtersClauses = buildPreparableFiltersClause(startIndex, propertyTypes, propertyTypeFilters)
     val filtersClause = if (filtersClauses.first.isNotEmpty()) " AND ${filtersClauses.first} " else ""
 
+    val metadataOptionColumns = metadataOptions.associateWith(::mapMetaDataToColumnSql)
+    val nonAggregatedMetadataSql = metadataOptionColumns
+            .filter { !isMetaDataAggregated(it.key) }.values.joinToString("")
```

Comment:
do these not need to be comma-separated?

Reply:
Nope, the `mapMetaDataToColumnSql` has already the comma prefix.